src/unistd.rs

```diff
@@ -756,6 +756,36 @@ pub fn gethostname<'a>(buffer: &'a mut [u8]) -> Result<&'a CStr> {
     })
 }
 
+/// Set the NIS domain name (see
+/// [setdomainname(2)](http://man7.org/linux/man-pages/man2/setdomainname.2.html)).
+#[cfg(target_os = "linux")]
```

Comment:
Both of these functions are available on FreeBSD too.  And while I haven't checked, I would bet that they're available on all of the BSDs based on their age.

Reply:
They're also declared on emscripten, so I'd add both on those platforms as well.

Reply:
I did a quick check and indeed they seem to be available on all BSD (plus ES) too.

Reply:
It looks like libc does not expose them, so https://github.com/rust-lang/libc/issues/868 needs to be fixed first.